tests/components/modbus/test_init.py

```diff
@@ -1407,9 +1408,71 @@ async def test_pb_service_write_no_slave(
     assert func_name[do_write[FUNC]].called
     assert func_name[do_write[FUNC]].call_args.args == (data[ATTR_ADDRESS],)
     assert func_name[do_write[FUNC]].call_args.kwargs == {
-        "slave": 1,
+        ATTR_DEVICE_ID: 1,
         value_arg_name[do_write[FUNC]]: data[do_write[DATA]],
     }
 
     if do_return[DATA]:
         assert any(message.startswith("Pymodbus:") for message in caplog.messages)
+
+
+@pytest.mark.parametrize(
+    "do_config",
+    [
+        {
+            CONF_NAME: TEST_MODBUS_NAME,
+            CONF_TYPE: SERIAL,
+            CONF_BAUDRATE: 9600,
+            CONF_BYTESIZE: 8,
+            CONF_METHOD: "rtu",
+            CONF_PORT: TEST_PORT_SERIAL,
+            CONF_PARITY: "E",
+            CONF_STOPBITS: 1,
+            CONF_SENSORS: [
+                {
+                    CONF_NAME: "dummy_noslave",
+                    CONF_ADDRESS: 8888,
+                }
+            ],
+        },
+    ],
+)
+@pytest.mark.parametrize(
+    "do_write",
+    [
+        {
+            DATA: ATTR_VALUE,
+            VALUE: 15,
+            SERVICE: SERVICE_WRITE_REGISTER,
+            FUNC: CALL_TYPE_WRITE_REGISTER,
+        },
+        {
+            DATA: ATTR_STATE,
+            VALUE: False,
+            SERVICE: SERVICE_WRITE_COIL,
+            FUNC: CALL_TYPE_WRITE_COIL,
+        },
+    ],
+)
+async def test_pb_service_return_None(
+    hass: HomeAssistant,
+    do_write,
+    caplog: pytest.LogCaptureFixture,
+    mock_modbus_with_pymodbus,
+) -> None:
+    """Run test when pymodbus returns None."""
+    func_name = {
```

Comment:
Okay so _what_ does this actually test? Like it returns None, but what does the user notice from this?

Reply:
It is called code coverage !!!

Reply:
No this is touching code, I am asking what it is testing, because okay cool pymodbus returns None, but what else should it return? What behavior do we expect when it returns None or what do we expect when it does not return None?

Reply:
it should return a modbus message cf https://github.com/pymodbus-dev/pymodbus/blob/9e22049439240a89f159a2aafe65137121201ce7/pymodbus/pdu/pdu.py#L12 

not sure it can return None in practise, because if there is no gateway, there is an exception returned
cf https://www.simplymodbus.ca/exceptions.htm

actually it was to cover those lines : https://github.com/home-assistant/core/blob/bf64e1196027f9cb6b4bb4ae9e8e43604751e8fe/homeassistant/components/modbus/modbus.py#L398

but I removed this "unfortunate" test

Reply:
well, I think it never return None in fact, but I am not sure.